cmd/spiffe-aws-assume-role/cli/cli.go

```diff
@@ -2,31 +2,44 @@ package cli
 
 import (
 	"fmt"
+	"time"
 
 	"github.com/alecthomas/kong"
 	"github.com/spiffe/go-spiffe/v2/spiffeid"
+	"github.com/square/spiffe-aws-assume-role/cmd/spiffe-aws-assume-role/cli/mappers"
 	"github.com/square/spiffe-aws-assume-role/pkg/credentials"
 	"github.com/square/spiffe-aws-assume-role/pkg/processcreds"
 )
 
 type CredentialsCmd struct {
-	Audience       string `required:"" help:"SVID JWT Audience. Must match AWS configuration"`
-	SpiffeID       string `required:"" help:"The SPIFFE ID of this workload"`
-	WorkloadSocket string `optional:"" help:"Path to SPIFFE Workload Socket"`
-	RoleARN        string `required:"" help:"AWS Role ARN to assume"`
-	SessionName    string `optional:"" help:"AWS Session Name"`
-	STSEndpoint    string `optional:"" help:"AWS STS Endpoint variable"`
+	Audience        string        `required:"" help:"SVID JWT Audience. Must match AWS configuration"`
+	SpiffeID        string        `required:"" help:"The SPIFFE ID of this workload"`
+	WorkloadSocket  string        `optional:"" help:"Path to SPIFFE Workload Socket"`
+	RoleARN         string        `required:"" help:"AWS Role ARN to assume"`
+	SessionName     string        `optional:"" help:"AWS Session Name"`
+	STSEndpoint     string        `optional:"" help:"AWS STS Endpoint variable"`
+	SessionDuration time.Duration `optional:"" type:"iso8601duration" help:"AWS session duration in ISO8601 duration format (e.g. PT5M for five minutes)"`
 }
```

Comment:
Although Kong is happy enough with the "bare" style of struct tag, go vet (which I did not know about) not so much. Excitingly, this issue was caught by our new CI workflow.